### ISSUE TITLE
Quarry unable to mine trough smoke and gas

### ISSUE THREAD

**Issue by oz-tal** [minetest-mods/technic#621]
when using mods like `smoke` and `mine_gas` (from `df_caverns`), gaseous blocks spreading in the mining area prevent digging.

It specifically occur [on this check](https://github.com/minetest-mods/technic/blob/master/technic/machines/HV/quarry.lua#L177):
```lua
if checknode.name ~= "air" then
	can_dig = false
	break
end
```

I hotfixed my server like this, but obviously an API to register unobstructives (but undiggables) blocks is required:
```lua
if checknode.name ~= "air" and checknode.name ~= "smoke:block" and checknode.name ~= "smoke:smog" and checknode.name ~= "mine_gas:gas"then
	can_dig = false
	break
end
```

It should be easy enough to implement, but I currently lack time to do it. I might very well come around it later with a PR, in the meantime I'm opening this issue.

**Comment by BuckarooBanzay** [comment]
Hm, the quarry does work very well with `vacuum:vacuum` nodes from the vacuum mod (https://github.com/mt-mods/vacuum/blob/master/vacuum.lua)

Haven't looked closer but i think this might just be an issue with the `drop` and `diggable` fields on the node-definition

**Comment by OgelGames** [comment]
@BuckarooBanzay this is `minetest-mods/technic`, the quarry hasn't been rewritten yet ;)

**Comment by BuckarooBanzay** [comment]
> @BuckarooBanzay this is `minetest-mods/technic`, the quarry hasn't been rewritten yet ;)

Oh god :facepalm: sorry for the noise, ignore my comments :/

**Comment by SmallJoker** [comment]
Adding a list of allowed nodes does not fix the issue for any other mods. Although, using a generalized way might accidentally match wrong nodes. Does this here work reliably in your case?

```Lua
local def = minetest.registered_nodes[checknode.name] or {}
local is_kind_of_gas = def.buildable_to and def.sunlight_propagates and not def.walkable
	and not def.diggable and (def.drawtype == "airlike" or def.drawtype == "glasslike")
if not is_kind_of_gas then
	can_dig = false
	break
end
```

**Comment by oz-tal** [comment]
Good fix, I've tested your solution and it's working very well

**Comment by SmallJoker** [comment]
Fixed as of fda8a3d

Please @ mention me if there is any similar or follow-up issue.
